components/elm/bld/ELMBuildNamelist.pm

```diff
@@ -3021,7 +3023,9 @@ sub setup_logic_fan {
             fatal_error('Cannot use_fan if use_crop is false');
           }   #
 
-          if( $nl_flags->{'use_fan'} eq ".true." ) {
+	  my $var = "use_fan";
+	  my $val = $nl->get_value($var);
+          if( $val eq ".true." ) {
```

Comment:
So it was a perl error?

Reply:
Yes. The information about `use_fan` was being extracted from the wrong data structure.